src/libraries/System.Private.CoreLib/src/System/Collections/Generic/NonRandomizedStringEqualityComparer.cs

```diff
@@ -16,13 +14,7 @@ namespace System.Collections.Generic
     // Needs to be public to support binary serialization compatibility
     public class NonRandomizedStringEqualityComparer : IEqualityComparer<string?>, IInternalStringEqualityComparer, ISerializable
     {
-        // Dictionary<...>.Comparer and similar methods need to return the original IEqualityComparer
-        // that was passed in to the ctor. The caller chooses one of these singletons so that the
-        // GetUnderlyingEqualityComparer method can return the correct value.
-
-        internal static readonly NonRandomizedStringEqualityComparer WrappedAroundDefaultComparer = new OrdinalComparer(EqualityComparer<string?>.Default);
-        internal static readonly NonRandomizedStringEqualityComparer WrappedAroundStringComparerOrdinal = new OrdinalComparer(StringComparer.Ordinal);
-        internal static readonly NonRandomizedStringEqualityComparer WrappedAroundStringComparerOrdinalIgnoreCase = new OrdinalIgnoreCaseComparer(StringComparer.OrdinalIgnoreCase);
+        private static readonly StringComparers s_stringComparers = new StringComparers();
```

Comment:
Is this workaround for some kind of JIT problem? It does not look like an improvement to me.

Reply:
Has two less calls to `[CORINFO_HELP_READYTORUN_STATIC_BASE]` though they'd likely be dropped at Tier1?

Reply:
Right. The extra type that this introduces costs a lot more than two calls to `[CORINFO_HELP_READYTORUN_STATIC_BASE]`.

Reply:
Reverted